### PR TITLE
docs: add EDK details and context around switching to utilities

### PR DESCRIPTION
**Pull request by pnadolny13**
Related to https://github.com/meltano/hub/issues/1115 and https://github.com/meltano/hub/pull/1135.

We didn't have anything in the docs about the EDK and why it exists. As I was writing this I thought giving the context would be helpful but let me know what you think.

Preview:

- https://deploy-preview-7239--meltano.netlify.app/

### CODE REVIEW

File: docs/src/_guide/advanced-topics.md [hunk 1]
```diff
@@ -14,6 +14,39 @@ Most of Meltano's features are available without installing any additional packa
 * `gcs` - Support for using Google Cloud Storage as a [state backend](/concepts/state_backends)
 * `azure` - Support for using Azure Blob Storage as a [state backend](/concepts/state_backends)
 
+## Extension Developer Kit (EDK)
+
+Meltano extensions are thin wrappers around applications that allow you to integrate existing data tools with Meltano.
```
Comment (tayloramurphy) [inline docs/src/_guide/advanced-topics.md:19]:
```suggestion
Meltano extensions are lightweight shims that allow you to integrate existing data tools with Meltano.
```

Thoughts on this language? I don't want to give the impression that we fully wrap the tool and that our Airflow is somehow different from a "pure" Airflow

File: docs/src/_guide/advanced-topics.md [hunk 2]
```diff
@@ -14,6 +14,39 @@ Most of Meltano's features are available without installing any additional packa
...
+Extensions allow the developer to add additional features like pre and post hooks to run before and after Meltano executes the wrapped application.
```
Comment (tayloramurphy) [inline docs/src/_guide/advanced-topics.md:20]:
```suggestion
Extensions allow the developer to add additional features like pre/post-hooks to run before/after Meltano executes the application, as well enabling project scaffolding to be customized for each plugin. This project scaffolding was previously accomplished via files bundles.
```

File: docs/src/_guide/advanced-topics.md [hunk 3]
```diff
@@ -14,6 +14,39 @@ Most of Meltano's features are available without installing any additional packa
...
+The Meltano [Extension Developer Kit](https://github.com/meltano/edk) (EDK) was created to make it easier for developers to build Meltano extensions.
+The EDK also has functionality to allow the extension to initialize the scaffold of a project directory which was traditionally the job of files bundles, ultimately simplifying the work needed for developers.
+For more information on how to build your own EDK-based Meltano extension, see [the EDK docs](https://edk.meltano.com/en/latest/).
```
Comment (tayloramurphy) [inline docs/src/_guide/advanced-topics.md:23]:
```suggestion

The Meltano [Extension Developer Kit](https://github.com/meltano/edk) (EDK) was created to make it easier for developers to build Meltano extensions.
For more information on how to build your own EDK-based Meltano extension, see [the EDK docs](https://edk.meltano.com/en/latest/).
```

File: docs/src/_guide/advanced-topics.md [hunk 4]
```diff
@@ -14,6 +14,39 @@ Most of Meltano's features are available without installing any additional packa
...
+* transformer (e.g. dbt-snowflake)
+* orchestrator (e.g. Airflow, Dagster)
+* utility (e.g. SqlFluff)
+
+
+The new approach is to group all non-EL plugins under the utility plugin type, leaving only the following plugin types:
+
+* extractor
+* loader
+* utility - now also including transformers and orchestrators
```
Comment (tayloramurphy) [inline docs/src/_guide/advanced-topics.md:42]:
```suggestion
* utility - now also including plugins previously called transformers and orchestrators
```

**Review by tayloramurphy — CHANGES_REQUESTED** [review]
@pnadolny13 love this. Minor edits

**Comment by pnadolny13** [comment]
@tayloramurphy thanks for the suggestions! I agree with you comment on the `shims` wording. Let me know if you have any other thoughts.

**Review by tayloramurphy — APPROVED** [review]
This looks good to me. I'm approving but let's hold off on merging until we coordinate with Engineering on the Hub PR as well and make sure they're aware and have reviewed.

**Comment by tayloramurphy** [comment]
@aaronsteers I'd like your review on this prior to merging. Target date of Friday merge.

File: docs/src/_guide/advanced-topics.md [hunk 5]
```diff
@@ -14,6 +14,39 @@ Most of Meltano's features are available without installing any additional packa
 * `gcs` - Support for using Google Cloud Storage as a [state backend](/concepts/state_backends)
 * `azure` - Support for using Azure Blob Storage as a [state backend](/concepts/state_backends)
 
+## Extension Developer Kit (EDK)
+
+Meltano extensions are lightweight shims that allow you to integrate existing data tools with Meltano.
```
Comment (aaronsteers) [inline docs/src/_guide/advanced-topics.md:19]: Shim can often mean something different. A different word here could be more accurate. Something like...

```suggestion
Meltano extensions are lightweight executables which allow you to integrate existing data tools with Meltano.
```

```suggestion
Meltano extensions are lightweight programs which allow you to integrate existing data tools with Meltano.
```

File: docs/src/_guide/advanced-topics.md [hunk 6]
```diff
@@ -14,6 +14,39 @@ Most of Meltano's features are available without installing any additional packa
...
+Extensions allow the developer to add additional features like pre/post-hooks to run before/after Meltano executes the application, as well enabling project scaffolding to be customized for each plugin. This project scaffolding was previously accomplished via files bundles.
+
+The Meltano [Extension Developer Kit](https://github.com/meltano/edk) (EDK) was created to make it easier for developers to build Meltano extensions.
+For more information on how to build your own EDK-based Meltano extension, see [the EDK docs](https://edk.meltano.com/en/latest/).
+
+### EDK Context
```
Comment (aaronsteers) [inline docs/src/_guide/advanced-topics.md:25]: This section is more about how Meltano is classifying plugins and less about the EDK itself - as the EDK itself is agnostic to the plugin type we place upon it.


```suggestion
### Meltano Plugin Types
```

Or something like

```suggestion
### Meltano Utilities
```

Or

```suggestion
### Selecting a Plugin Type
```

Or

```suggestion
### Understanding Plugin Types
```

**Review by aaronsteers — APPROVED** [review]
A few notes. Feel free to resolve as you see fit and merge if nothing else outstanding. Thanks!

**Pull request merged**